components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java

```diff
@@ -290,6 +303,32 @@ public boolean globallyEngageModule(String moduleId) throws ModuleMgtException {
 
     }
 
+    /**
+     * This method persists the globally engaged status of a module in  the registry
+     * @param axisModule
+     * @param globallyEngagedStatus
+     */
+    private void persistGloballyEngagedStatus(AxisModule axisModule, boolean globallyEngagedStatus) {
+        String moduleResourcePath = getModuleResourcePath(axisModule);
+        Resource moduleResource;
+        try {
+            if (registry.resourceExists(moduleResourcePath)) {
+                moduleResource = registry.get(moduleResourcePath);
+            } else {
+                moduleResource = registry.newResource();
+            }
+            moduleResource.setProperty(RegistryResources.ModuleProperties.GLOBALLY_ENGAGED,
+                                       Boolean.toString(globallyEngagedStatus));
+            registry.put(moduleResourcePath, moduleResource);
+        } catch (RegistryException e) {
+            log.error("Failed to persist globally engaged status of the module: " + axisModule.getName(), e);
```

Comment:
Thanks Nirmal for pointing the issue. Fixed with https://github.com/wso2/carbon-deployment/pull/146